GetStream/stream-chat-android
# CAS-900 Implement default Channel swipe actions behaviours

https://stream-io.atlassian.net/browse/CAS-900

### 🎯 Goal
So far, channel options were shown to the user (when swiping channel item view) without default implementation. We shouldn't show clickable icons that do nothing by default.

### 🛠 Implementation details

1. Add default Channel "delete" option implementation coherent to default MessageListView's delete message implementation (showing confirmation dialog).
2. Make Channel "more" option not displayed by default since we're not able to provide any default implementation.

### 🎨 UI Changes



### 🧪 Testing
When you:
1. Remove `ChannelListView::configureDefaultMoreOptionsListener` invocation, 
2. Remove `app:streamUiChannelOptionsEnabled="true"` attribute from `fragment_channels.xml`, 
3. Remove `ChannelListView::setChannelDeleteClickListener` call,
and try out default implementation of Channel options provided by SDK.

Then:
1. "More options" should not be shown by default. 
2. "Delete channel" option should be shown only for the channel the current user is privileged to delete.
3. In case "More options" and "Delete channel" options are not available for a specific channel item swipe behavior should be disabled.

### ☑️ Checklist


### 🎉 GIF

![swipe](https://user-images.githubusercontent.com/4527432/119818039-fe20fa00-beee-11eb-8171-3cf47c82e023.gif)
